File: src/main/scala/SparkWeek11/wordCount.scala
package SparkWeek11

import org.apache.log4j.{Level, Logger}
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._
import org.apache.spark._

object wordCount extends App{

  Logger.getLogger("org").setLevel(Level.ERROR)
  val sc = new SparkContext("local[*]","Count")  //-- this is for local run

  val spark = SparkSession.builder()
    .appName("WordCount")
    .master("local[*]")
    .getOrCreate()

  val rdd1 = sc.textFile("src/main/resources/wordCount.txt")
  val rdd2 = rdd1.flatMap(x => x.split(" "))
  val rdd3 = rdd2.map(x=>(x,1))
  val rdd4 = rdd3.reduceByKey((x,y) => (x+y))
  rdd4.collect.foreach(println)

}

File: src/main/pySpark/word_Count.py
from pyspark import SparkContext
from pyspark.sql import *
from pyspark.sql.functions import *

spark = SparkSession.builder \
    .master("local[2]") \
    .appName("word_count") \
    .getOrCreate()

sc = spark.sparkContext

rdd1 = sc.textFile(r"C:\Users\Manish\CodeSparkScala\src\main\resources\wordCount.txt")


rdd2 = rdd1.flatMap(lambda x:x.split(" "))


rdd3 = rdd2.map(lambda x:(x,1))


rdd4 = rdd3.reduceByKey(lambda x,y : x+y)

rdd4.foreach(print)
rdd4.saveAsTextFile(r"C:\Users\Manish\CodeSparkScala\src\main\resources\OutwordCount.txt")